ext/standard/image.c

```diff
@@ -1353,13 +1475,23 @@ PHPAPI int php_getimagetype(php_stream * stream, const char *input, char *filety
 	if (php_get_wbmp(stream, NULL, 1)) {
 		return IMAGE_FILETYPE_WBMP;
 	}
-    if (!twelve_bytes_read) {
+
+	php_gd_image_reader reader = { .stream = stream, .data = NULL, .data_pos = 0 };
+
+
+	if (!php_stream_rewind(stream) && php_is_image_avif(&reader)) {
+		return IMAGE_FILETYPE_AVIF;
+	}
```

Comment:
Does it need to be? The current way has worked for me. On the other hand... I see your point; `php_get_wbmp()` never checks to see if the stream it's passed is actually a WBMP image. It just assumes that it is. So if it mistakenly found a valid height and width in a non-WBMP file, it would cheerily return a false positive.

I've moved the AVIF check up!